scipy/optimize/_differentialevolution.py

```diff
@@ -480,41 +622,85 @@ def solve(self):
 
         # do the optimisation.
         for nit in range(1, self.maxiter + 1):
+        
             if self.dither is not None:
                 self.scale = self.random_number_generator.rand(
                 ) * (self.dither[1] - self.dither[0]) + self.dither[0]
-            for candidate in range(np.size(self.population, 0)):
+                
+            # brake the population to subpopulation depending on the size of the pool
+            # determine number of sub-populations 'nsp', or number of 
+            # parallel evaluations, which is 'nsp + 1'
+            # and the length of the reminder 'lenrem'
+            nsp, lenrem = divmod(np.size(self.population, 0), self.poolsize)
+            
+            itsp = 0
+            # iterate among sub-populations
+            # when self.poolsize == 1 the mutation is aggressive
+            # when self.poolsize > 1 the mutation is quasi-aggressive
+            while itsp <= nsp:
+                # determine the length of the subpopulation
+                lensp = self.poolsize if (itsp < nsp) else lenrem
+
+                # initialize list for all members (parameters) of the current subpopulation
+                spparams = []
+                # mutate parameters for the sub-population
+                for candidate in range(lensp):
+                    trial = self._mutate(candidate + self.poolsize*itsp)
+                    self._ensure_constraint(trial)
+                    spparams.append(self._scale_parameters(trial))
+
+                    nfev += 1
+
+                # in parallel case the self.func must return a list of energies
+                # for the whole subpopulation
+                if len(self.args) is not 0:
```

Comment:
Replace lines 656 to lines 659 by:

    # params.shape = (lensp, self.num_params) and is iterable over the rows
    spenergies = self.pool_map(unpacker(self.func), params, itertools.repeat(args, lensp))